src/v/pandaproxy/client/client.cc

```diff
@@ -134,4 +138,86 @@ ss::future<kafka::produce_response::partition> client::produce_record_batch(
     return _producer.produce(std::move(tp), std::move(batch));
 }
 
+kafka::fetch_response::partition
+make_fetch_response(model::topic_partition tp, std::exception_ptr ex) {
+    kafka::error_code error;
+    try {
+        std::rethrow_exception(std::move(ex));
+    } catch (const partition_error& ex) {
+        vlog(ppclog.debug, "handling partition_error {}", ex.what());
```

Comment:
i think you can pass `ex` directly and `fmt` will / seastar-logging will invoke `what` automatically.